src/source/Ice/TurnConnection.c

```diff
@@ -895,6 +896,8 @@ STATUS turnConnectionStepState(PTurnConnection pTurnConnection)
 
                 pTurnConnection->state = TURN_STATE_GET_CREDENTIALS;
                 pTurnConnection->stateTimeoutTime = currentTime + DEFAULT_TURN_GET_CREDENTIAL_TIMEOUT;
+            } else {
+                CHK(currentTime < pTurnConnection->stateTimeoutTime, STATUS_TURN_CONNECTION_STATE_TRANSITION_TIMEOUT);
```

Comment:
Can we hit this in a unit test?

Reply:
I can't. I did manually tested it by overriding socketConnectionIsConnected to always return false.